books/free-programming-books-langs.md

```diff
@@ -1275,13 +1275,16 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Introduction to Node.js](https://nodejs.dev/learn) (HTML)
 * [Mastering Node](https://github.com/visionmedia/masteringnode) - visionmedia ([PDF](https://github.com/visionmedia/masteringnode/blob/master/book.pdf))
 * [Mixu's Node Book](http://book.mixu.net/node/)
-* [Node Documentation](https://nodejs.org/en/docs/)
+* [Node Design Patterns](https://ia801309.us.archive.org/5/items/HandbookOfNeuralComputingApplicationsPDFStormRG/Node.js%20Design%20Patterns%20-%20Casciaro,%20Mario%20%5BPDF%5D%5BStormRG%5D.pdf) - Mario Casciaro (PDF)
+* [Node Documentation](https://nodejs.org/en/docs/) (PDF)
+* [Node For Embedded System](https://www.pdfdrive.com/nodejs-for-embedded-systems-using-web-technologies-to-build-connected-devices-e157056110.html) - Pactrick Mulder and Kelsey Breseman (PDF)
```

Comment:
```suggestion
```
In general, we do not link to file sharing sites, or file hosting sites such as google docs. or dropbox.